src/realm/utilities.hpp

```diff
@@ -227,22 +227,42 @@ struct is_any<T, U, Ts...> : is_any<T, Ts...> {
 };
 
 
-// Use safe_equal() instead of std::equal() when comparing sequences which can have a 0 elements.
+// Use safe_equal() instead of std::equal() if one of the parameters can be a null pointer.
 template <class InputIterator1, class InputIterator2>
 bool safe_equal(InputIterator1 first1, InputIterator1 last1, InputIterator2 first2)
 {
-#if defined(_MSC_VER) && defined(_DEBUG)
-
-    // Windows has a special check in debug mode against passing realm::null()
-    // pointer to std::equal(). It's uncertain if this is allowed by the C++ standard. For details, see
+#if defined(_MSC_VER)
+    // VS has a special check in debug mode against passing a null pointer std::equal(); it will give a warning
+    // at runtime if this is observed.
+    // It's uncertain if this is allowed by the C++ standard. For details, see
     // http://stackoverflow.com/questions/19120779/is-char-p-0-stdequalp-p-p-well-defined-according-to-the-c-standard.
-    // Below check 'first1==last1' is to prevent failure in debug mode.
-    return (first1 == last1 || std::equal(first1, last1, first2));
+    // So we use a safe C++14 method instead that takes two range pairs.
+    return std::equal(first1, last1, first2, first2 + (last1 - first1));
```

Comment:
For readability, I think it would be nice to have `InputIterator2 last2 = first2 + (last1 - first1)` extracted.